### ISSUE TITLE
Several suggestions

### ISSUE THREAD

**Issue by smathot** [JdenHartog/vu_ams#1]
### Properly registering the `ams` object

Right now, you're setting the `ams` object as a property of `self.var` (and `self.experiment.var`). That's not the right way (it's not an experimental variable), and it will have weird side effects. It's better to attach the `ams` to the `experiment` object, and then register it in the Python workspace so that users can refer to it directly as `ams` (i.e. no `self.` prefix necessary). Like so:

```
# Dynamically create an ams instance
if not hasattr(self.experiment, u'ams'):
    self.experiment.ams = create_ams_instance() # Whatever needs to happen here
    self.experiment.cleanup_functions.append(self.close)
    self.python_workspace[u'ams'] = self.experiment.ams
```

Does that make sense?
### Using the var store

Instead of `self.experiment.get(u'my_var')` (which triggers `DeprecationWarning`s) you can use `self.experiment.var.my_var`. And the same for `self.experiment.set()`. In other words, the experiment has its own var store, although you probably don't need to use it, see the point above.
### Create a setup.py and upload to PyPi

It's pretty easy to create a setup script, and then upload the plugin to PyPi. Once you've done that, users can simply `pip install` the plugin in OpenSesame. See:
- http://osdoc.cogsci.nl/3.1/dev/plugin/#writing-a-setuppy-and-uploading-to-pypi
- https://github.com/smathot/opensesame-plugin-example

**Comment by JdenHartog** [comment]
Thanks for the suggestions! Some followup questions bellow if I may  :blush:

### Properly registering the ams object
- This definitely makes sense. I'm working on it. :wrench: I'm also including a dummy class to be able to use the plug-in inline ams.function() in debug mode with no hardware connected.

### Using the var store
- I'm a bit doubtful on using `self.var.my_var` or `self.experiment.var.my_var`.
  Is `self.var.my_var` item level and `self.experiment.var.my_var` experiment level?
- I'm using `self.experiment.set(u'marker_sent_time_%s' % self.name, str(self.time()))` to be able to log the marker_sent_time_[item name]. I have no idea if this is possible using something like `self.var.marker_sent_time_[item name] = self.time()` or do I have to stick to getters and setters for this?
- I tried to see the DeprecationWarnings using the _--warnings_ command-line argument as a way off checking the plug-in for correctness but `self.experiment.set(u'my_var',1)` didn't trigger any warnings.

### Create a setup.py and upload to PyPi
- I'll postpone this for now and look into this later  :alarm_clock:

**Comment by smathot** [comment]
> I'm a bit doubtful on using self.var.my_var or self.experiment.var.my_var.
> Is self.var.my_var item level and self.experiment.var.my_var experiment level?

Exactly. So you would store settings of the plugin in `self.var`. If you want to store some variable that should be generally accessible, such as response time or something, you would set it with `self.experiment.var`.

> I'm using self.experiment.set(u'marker_sent_time_%s' % self.name, str(self.time())) to be able to log the marker_sent_time_[item name]. I have no idea if this is possible using something like self.var.marker_sent_time_[item name] = self.time() or do I have to stick to getters and setters for this?

If you don't know the variable name beforehand, you indeed have to use the getters and setters, but the ones from the `var` object. So:

``` .python
self.experiment.var.set(u'marker_sent_time_%s' % self.name, self.clock.time())
```

> I tried to see the DeprecationWarnings using the --warnings command-line argument as a way off checking the plug-in for correctness but self.experiment.set(u'my_var',1) didn't trigger any warnings.

They're there somewhere. But, in any case, best to use `var` and avoid DeprecationWarnings altogether!
